subprojects/logging/src/main/java/org/gradle/internal/logging/events/RenderableOutputEvent.java

```diff
@@ -16,13 +16,16 @@
 
 package org.gradle.internal.logging.events;
 
+import org.gradle.api.Nullable;
 import org.gradle.api.logging.LogLevel;
 import org.gradle.internal.logging.text.StyledTextOutput;
 
 public abstract class RenderableOutputEvent extends CategorisedOutputEvent {
+    private final OperationIdentifier operationId;
```

Comment:
@bmuschko We shouldn't be using category to determine the type of operation. This is what the operation's descriptor is for.

We're not accepting workarounds for missing information in this work, and trying to extract something from the category is very much a workaround. We should instead add information to the progress events to explicitly say what type the operation has. Could be just a simple enum, doesn't need to be complex.